source/common/stats/tag_producer_impl.cc

```diff
@@ -101,36 +101,41 @@ void TagProducerImpl::forEachExtractorMatching(
 
 std::string TagProducerImpl::produceTags(absl::string_view metric_name, TagVector& tags) const {
   // TODO(jmarantz): Skip the creation of string-based tags, creating a StatNameTagVector instead.
-  tags.insert(tags.end(), default_tags_.begin(), default_tags_.end());
+  // tags.insert(tags.end(), default_tags_.begin(), default_tags_.end());
   IntervalSetImpl<size_t> remove_characters;
   TagExtractionContext tag_extraction_context(metric_name);
   std::vector<absl::string_view> tokens;
-  forEachExtractorMatching(metric_name, [&remove_characters, &tags, &tag_extraction_context](
-                                            const TagExtractorPtr& tag_extractor) {
-    tag_extractor->extractTag(tag_extraction_context, tags, remove_characters);
+  absl::flat_hash_set<absl::string_view> dup_set;
+  forEachExtractorMatching(metric_name, [&remove_characters, &tags, &tag_extraction_context,
+                                         &dup_set](const TagExtractorPtr& tag_extractor) {
+    bool other_extractor_with_same_name_exists = tag_extractor->otherExtractorWithSameNameExists();
+    if (other_extractor_with_same_name_exists &&
+        dup_set.find(tag_extractor->name()) != dup_set.end()) {
+      ENVOY_LOG_MISC(error, "Skipping duplicate tag for ", tag_extractor->name());
```

Comment:
This could get extremely noisy/verbose. Maybe this should be one of the every-so-often log macros?

Reply:
I think it won't happen that often because once a stat is populated it won't be re-extracted. However I switched to the power-of-2 logger just in case.